src/Map/doc/index.rst

```diff
@@ -0,0 +1,285 @@
+
+Available providers
+-------------------
+
+Leaflet
+~~~~~~~
+
+You can use `Leaflet`_ as the default provider by configuring the ``UX_MAP_DSN`` environment variable like this:
+
+.. code-block:: env
+
+    # .env
+    UX_MAP_DSN=leaflet://default
+
+Then, enable the Leaflet`_ Stimulus controller in your ``assets/controllers.json``:
+
+.. code-block:: json
+    
+    {
+        "controllers": {
+            "@symfony/ux-map": {
+                "leaflet": {
+                    "enabled": true,
+                    "fetch": "lazy"
+                }
+            },
+        },
+        "entrypoints": []
+    }
+
+You can configure the map with specific Leaflet options by passing a ``LeafletOptions`` instance to the ``options`` method::
+
+    use Symfony\UX\Map\Provider\Leaflet\LeafletOptions;
+    use Symfony\UX\Map\Provider\Leaflet\Option as LeafletOption;
+
+    $map
+        ->options(new LeafletOptions(
+            tileLayer: new LeafletOption\TileLayer(
+                url: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png',
+                attribution: '© OpenStreetMap contributors',
+                options: [
+                    'maxZoom' => 19,
+                ]
+            ),
+        ));
+        
+If everything went well, you should see a Leaflet map in your page.
+
+Google Maps
+~~~~~~~~~~~
+
+You can use `Google Maps`_ as the default provider by configuring the ``UX_MAP_DSN`` environment variable like this:
+
+.. code-block:: env
+
+    # .env
+    UX_MAP_DSN=google-maps://GOOGLE_MAPS_API_KEY@default
+    UX_MAP_DSN=google-maps://GOOGLE_MAPS_API_KEY@default?version=weekly
+    UX_MAP_DSN=google-maps://GOOGLE_MAPS_API_KEY@default?language=fr&region=FR
+
+The provider has a number of options:
+
+======================  ======================================  ===================================
+     Option             Description                             Default
+======================  ======================================  ===================================
+``id``                  The id of the script tag                __googleMapsScriptId
+``language``            Force language, see                     The user's preferred language 
+                        `list of supported languages`_          specified in the browser
+``region``              Unicode region subtag identifiers
+                        compatible with `ISO 3166-1`_         
+``nonce``               Use a cryptographic nonce attribute        
+``retries``             The number of script load retries       3
+``url``                 Custom url to load the Google Maps API  https://maps.googleapis.com/maps/api/js
+                        script       
+``version``             The release channels or version         weekly
+                        numbers                       
+======================  ======================================  ===================================
```

Comment:
Better set advanced configuration _after_ the "then enable the Stimulus controller" i think